engine/runtime-compiler/src/main/java/org/enso/compiler/data/BindingsMapBase.java

```diff
@@ -129,5 +147,58 @@ final State withCurrentModule(ModuleReference newModule) {
     final State withExportedSymbols(Map<String, List<ResolvedName>> newSymbols) {
       return new State(definedEntities, currentModule, resolvedImports, newSymbols);
     }
+
+    public List<DefinedEntity> definedEntities() {
+      return definedEntities;
+    }
+
+    public ModuleReference currentModule() {
+      return currentModule;
+    }
+
+    public List<ResolvedImport> resolvedImports() {
+      return resolvedImports;
+    }
+
+    public Map<String, List<ResolvedName>> exportedSymbols() {
+      return exportedSymbols;
+    }
+
+    final List<? extends ResolvedName> findQualifiedImportCandidates(String name) {
+      if (cacheImportsByName == null) {
+        var ribn = new TreeMap<String, List<ImportTarget>>();
+        resolvedImports
+            .flatMap(i -> allNames(i))
+            .foreach(
+                (n) -> {
+                  return ribn.computeIfAbsent(n, this::findQualifiedImportCandidatesImpl);
+                });
+        cacheImportsByName = ribn;
+      }
+      return cacheImportsByName.getOrDefault(name, nil());
+    }
+
+    private List<ImportTarget> findQualifiedImportCandidatesImpl(String name) {
+      return resolvedImports
+          .filter(i -> importMatchesName(i, name) && !i.isSynthetic())
```

Comment:
- this traversal thru `resolvedImports` list must be pretty expensive
- can be speed things up by `cacheImportsByName`?
- [startup benchmark](https://github.com/enso-org/enso/actions/runs/16312653003) is running